v11y/api/src/db/mod.rs

```diff
@@ -377,9 +402,87 @@ impl Db {
             .bind(id)
         };
 
+        query
+            .fetch(&self.pool)
+            .filter_map(|row| async move { row.ok().map(|row| (row.get("origin"), row.get("related"))) })
+    }
+
+    pub async fn get_references<'s>(
+        &'s self,
+        id: &'s str,
+        r#type: Option<String>,
+        origin: Option<String>,
+    ) -> impl Stream<Item = (String, Reference)> + 's {
+        let query = if let Some(origin) = origin {
+            if let Some(ty) = r#type {
+                sqlx::query(
+                    r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1 and origin = $2 and type = $3
+                "#,
+                )
+                .bind(id)
+                .bind(origin)
+                .bind(ty)
+            } else {
+                sqlx::query(
+                    r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1 and origin = $2
+                "#,
+                )
+                .bind(id)
+                .bind(origin)
+            }
+        } else if let Some(ty) = r#type {
+            sqlx::query(
+                r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1 and type = $2
+                order by
+                    origin
+                "#,
+            )
+            .bind(id)
+            .bind(ty)
+        } else {
+            sqlx::query(
+                r#"
+                select
+                    origin, type, url
+                from
+                    refs
+                where
+                    vulnerability_id = $1
+                order by
+                    origin
+                "#,
+            )
+            .bind(id)
+        };
```

Comment:
Again, possibly a style thing, but IMO, this would be much clearer as a "tuple match" than nested if's...

```suggestion
        let query = match (origin, r#type) {
            (Some(origin), Some(ty)) => ...,
            (Some(origin), None) => ...,
            (None, Some(ty)) => ...,
            (None, None) => ...,
        };
```

Reply:
Yep, thanks!